MekHQ/src/mekhq/gui/adapter/PersonnelTableMouseAdapter.java

```diff
@@ -1904,81 +1905,95 @@ private void maybeShowPopup(MouseEvent e) {
-                    //Start of role-specific Edge reroll options
-                    //Mechwarriors
-                    if (person.getPrimaryRole() == Person.T_MECHWARRIOR) {
-                        cbMenuItem = new JCheckBoxMenuItem(resourceMap.getString("edgeTriggerHeadHits.text")); //$NON-NLS-1$
-                        cbMenuItem.setSelected(person.getOptions()
-                                .booleanOption(OPT_EDGE_HEADHIT));
-                        cbMenuItem.setActionCommand(makeCommand(CMD_EDGE_TRIGGER, OPT_EDGE_HEADHIT));
-                        cbMenuItem.addActionListener(this);
-                        menu.add(cbMenuItem);
-                        cbMenuItem = new JCheckBoxMenuItem(
-                                resourceMap.getString("edgeTriggerTAC.text")); //$NON-NLS-1$
-                        cbMenuItem.setSelected(person.getOptions()
-                                .booleanOption(OPT_EDGE_TAC));
-                        cbMenuItem.setActionCommand(makeCommand(CMD_EDGE_TRIGGER, OPT_EDGE_TAC));
-                        cbMenuItem.addActionListener(this);
-                        menu.add(cbMenuItem);
-                        cbMenuItem = new JCheckBoxMenuItem(resourceMap.getString("edgeTriggerKO.text")); //$NON-NLS-1$
-                        cbMenuItem.setSelected(person.getOptions()
-                                .booleanOption(OPT_EDGE_KO));
-                        cbMenuItem.setActionCommand(makeCommand(CMD_EDGE_TRIGGER, OPT_EDGE_KO));
-                        cbMenuItem.addActionListener(this);
-                        menu.add(cbMenuItem);
-                        cbMenuItem = new JCheckBoxMenuItem(resourceMap.getString("edgeTriggerExplosion.text")); //$NON-NLS-1$
-                        cbMenuItem.setSelected(person.getOptions()
-                                .booleanOption(OPT_EDGE_EXPLOSION));
-                        cbMenuItem.setActionCommand(makeCommand(CMD_EDGE_TRIGGER, OPT_EDGE_EXPLOSION));
-                        cbMenuItem.addActionListener(this);
-                        menu.add(cbMenuItem);
-                        cbMenuItem = new JCheckBoxMenuItem(resourceMap.getString("edgeTriggerMASCFailure.text")); //$NON-NLS-1$
-                        cbMenuItem.setSelected(person.getOptions()
-                                .booleanOption(OPT_EDGE_MASC_FAILURE));
-                        cbMenuItem.setActionCommand(makeCommand(CMD_EDGE_TRIGGER, OPT_EDGE_MASC_FAILURE));
-                        cbMenuItem.addActionListener(this);
-                        menu.add(cbMenuItem);
-                    //Doctors    
-                    } else if (person.getPrimaryRole() == Person.T_DOCTOR && gui.getCampaign().getCampaignOptions().useSupportEdge()) {
+                    //Start of Edge reroll options
+                    cbMenuItem = new JCheckBoxMenuItem(resourceMap.getString("edgeTriggerHeadHits.text")); //$NON-NLS-1$
```

Comment:
This five-liner block really wants to be turned into a helper function that takes some parameters (the text, the edge trigger, list of roles) and returns a JCheckBoxMenuItem ready for addition to the menu.

Reply:
It's practically begging, but I had my headset on with some loud music so I didn't hear it at first.